### PR TITLE
Removing bundle manager and registering bundles in extensions.php

### LINKED ISSUE #11633: [Feature]: Remove Extension / Bundle/Brick Manager
### Feature description

Bundles should be manually registered in `App\Kernel` or using Symfony Flex `bundle.php` no more Pimcore magic. 

#12274

### PR DESCRIPTION
**Pull request by aashan10**
## Changes in this pull request  
Resolves #11633

### CODE REVIEW

File: lib/Document/Editable/EditableHandler.php [hunk 1]
```diff
@@ -149,6 +149,7 @@ public function __construct(
     }
 
     /**
+     * @deprecated With the removal of extension config states, this is no longer of use
```
Comment (jdreesen) [inline lib/Document/Editable/EditableHandler.php]: Can we deprecate this in `10.5.x` and remove it in `11.0`?

File: lib/Extension/Document/Areabrick/AreabrickManagerInterface.php [hunk 2]
```diff
@@ -58,35 +58,4 @@ public function getBricks(): array;
...
-     */
-    public function isEnabled(string $id): bool;
-
-    /**
-     * Enables/disables an areabrick
-     *
-     * @param string $id
-     * @param bool $state
-     */
-    public function setState(string $id, bool $state);
```
Comment (jdreesen) [inline lib/Extension/Document/Areabrick/AreabrickManagerInterface.php:91]: Should we add the `@deprecated` annotations to the interface in `10.5.x`, too? Currently, it's only in the concrete implementation.

**Comment by aashan10** [comment]
@jdreesen I have created a followup pr considering the changes you requested for 10.5.

File: bundles/InstallBundle/src/Installer.php [hunk 3]
```diff
@@ -755,7 +754,6 @@ protected function insertDatabaseContents()
             'http_errors',
             'notes_events',
             'objects',
-            'plugins', // TODO: to be removed in Pimcore 11
```
Comment (dvesh3) [inline bundles/InstallBundle/src/Installer.php:758]: we should also add a migration to remove this from `users_permission_definitions` table.

File: lib/Extension/Bundle/PimcoreBundleInterface.php [hunk 4]
```diff
@@ -49,15 +49,6 @@ public function getVersion();
...
     public function getInstaller();
 
-    /**
-     * @deprecated will be removed in Pimcore 11
-     *
-     * Get path to include in admin iframe
-     *
-     * @return string|RouteReferenceInterface|null
-     */
-    public function getAdminIframePath();
```
Comment (dvesh3) [inline lib/Extension/Bundle/PimcoreBundleInterface.php:59]: Please remove it from Abstract class https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/lib/Extension/Bundle/AbstractPimcoreBundle.php#L57

Also docs https://github.com/pimcore/pimcore/blob/0bb752535b36b5d891e1b67edff3ee4caaca09f1/doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/05_Pimcore_Bundles/README.md#L33

File: lib/Extension/Bundle/Config/StateConfig.php [hunk 5]
```diff
@@ -1,266 +0,0 @@
...
- * - GNU General Public License version 3 (GPLv3)
- * - Pimcore Commercial License (PCL)
- * Full copyright and license information is available in
- * LICENSE.md which is distributed with this source code.
- *
- *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
- *  @license    http://www.pimcore.org/license     GPLv3 and PCL
- */
-
-namespace Pimcore\Extension\Bundle\Config;
```
Comment (dvesh3) [inline lib/Extension/Bundle/Config/StateConfig.php:18]: Remove it from here https://github.com/pimcore/pimcore/blob/b2110bb3c416cbf4a7fce967fe98a0e031d642a0/bundles/CoreBundle/config/extensions.yaml#L26

File: lib/Extension/Bundle/Traits/StateHelperTrait.php [hunk 6]
```diff
@@ -27,39 +27,13 @@
...
      * Helper method to check installed state from class definitions/non-service instances
      *
      * @return bool
      */
     public static function isInstalled(): bool
     {
-        if (!static::isEnabled()) {
-            return false;
-        }
-
```
Comment (dvesh3) [inline lib/Extension/Bundle/Traits/StateHelperTrait.php:62]: Should we move `isInstalled()` method to the `AbstractPimcoreBundle` class, which in turn injects the bundle manager?

**Review by dvesh3 — COMMENTED** [review]
please also create a PR on pimcore/demo to register bundles through `bundles.php`instead of https://github.com/pimcore/demo/blob/11.x/var/config/extensions.php

File: lib/Extension/Bundle/AbstractPimcoreBundle.php [hunk 7]
```diff
@@ -90,4 +84,18 @@ public function getEditmodeCssPaths()
...
+            self::$bundleManager = \Pimcore::getContainer()->get(PimcoreBundleManager::class);
+        }
+
+        $bundle = self::$bundleManager->getActiveBundle(__CLASS__, false);
+        if (!self::$bundleManager->isInstalled($bundle)) {
+            return false;
+        }
+
+        return true;
+    }
```
Comment (jdreesen) [inline lib/Extension/Bundle/AbstractPimcoreBundle.php]: I think we need late static binding here. In the trait it was fine, but we want the values from the subclass, not from `AbstractPimcoreBundle`, right?

```suggestion
    public static function isInstalled(): bool
    {
        static::$bundleManager ??= \Pimcore::getContainer()->get(PimcoreBundleManager::class);

        $bundle = static::$bundleManager->getActiveBundle(static::class, false);

        return static::$bundleManager->isInstalled($bundle);
    }
```

File: bundles/CoreBundle/src/Command/Bundle/ListCommand.php [hunk 8]
```diff
@@ -75,21 +67,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-
-                $bundleState = $this->bundleManager->getState($bundle);
-                $row[] = $bundleState['priority'];
-            } else {
-                $row[] = false;
-                $row[] = false;
-                $row[] = false;
-                $row[] = 0;
-            }
+            $row[] = $this->bundleManager->isInstalled($bundleClass);
```
Comment (dvesh3) [inline bundles/CoreBundle/src/Command/Bundle/ListCommand.php]: Error on executing command
image

Comment (dvesh3) [reply bundles/CoreBundle/src/Command/Bundle/ListCommand.php]: fixed

File: lib/Extension/Bundle/PimcoreBundleManager.php [hunk 9]
```diff
@@ -222,17 +201,59 @@ private function getManuallyRegisteredBundleState(): array
...
+        $resolver->setAllowedTypes('environments', 'array');
+
+        $resolver->setNormalizer('environments', function (Options $options, $value) {
+            // normalize to string and trim
+            $value = array_map(function ($item) {
+                $item = (string)$item;
+                $item = trim($item);
+
+                return $item;
+            }, $value);
```
Comment (jdreesen) [inline lib/Extension/Bundle/PimcoreBundleManager.php]:
```suggestion
            $value = array_map(fn ($item) => trim((string) $item), $value);
```

File: lib/Extension/Bundle/PimcoreBundleManager.php [hunk 10]
```diff
@@ -222,17 +201,59 @@ private function getManuallyRegisteredBundleState(): array
...
+
+            // remove empty values
+            return array_filter($value, function ($item) {
+                return !empty($item);
+            });
```
Comment (jdreesen) [inline lib/Extension/Bundle/PimcoreBundleManager.php]:
```suggestion
            return array_filter($value, fn ($item) => !empty($item));
```

**Review by dvesh3 — COMMENTED** [review]
please get rid of translations (from `en.extended.json` only). which are quite specific to deleted interfaces. e.g.
- please_dont_forget_to_reload_pimcore_after_modifications
- extension_manager_state_change_not_allowed
- clear_cache_and_reload

**Pull request merged**